Website Optimization/Testing and analyzing.md

```diff
@@ -0,0 +1,101 @@
+
+For a good reference point, [GlobalStats](http://gs.statcounter.com/) gives up-to-date statistics on market share for operating systems, browser usage, and even what screen size is being used to view pages. Additionally, if the client can provide their own usage analytics, the specifications can be narrowed down even further.
+
+Advanced options like selecting different mobile devices (iPhone, Samsung, Google Pixel, and more), mobile browsers (Chrome, Firefox, Safari...), along with bandwidth throttling options (2G/3G/4G) are usually additional features that require creating an account or paying subscription.
+
+### How to read data?
+
+Running tests is fairly simple. Entering URL is usually all that needs to be done. After a few seconds of analyzing, results can be read, sometimes saved, or downloaded.
+
+PageSpeed Insights returns a few metrics marked as critical or vital. If you know what they represent, you can pull enough conclusions from them to plan your next move.
+
+For example, if _Time to Interactive_ is the amount of time it takes for the page to become fully interactive. This allows us to visualize in phase of the page loading process this occurs and what it means for the end-user.
+
+Another example is _Largest Contentful Paint_ - the time at which the largest text or image element is displayed ("painted"). This metric could indicate a bottleneck or something that requires extra attention (if results aren't satisfying, of course).
+
+![PageSpeed Insights - metrics](/img/pagespeed-metrics.png)
+
+GTmetrix provides a much more detailed report than PageSpeed Insights. All data is organized in report tabs. Under the _structure_ tab, all issues are displayed ordered by priority, from high to low. More information about each item can be seen by clicking on it.
+
+In this case, there are not many critical areas. Just an excessive DOM size warning and a recommendation for using a CDN. All other issues don't have too much impact on overall performance, but they still could be points of improvement. When items are expanded, GTmetrix gives a nice explanation for the issue along with technical details, so everything is on track and easy to understand.
+
+![GTmetrix structure](/img/gtmetrix-structure.png)
+
+### Mobile vs. desktop
+
+When running a speed test for a mobile device and a desktop device, you could notice quite the difference in performance results. This can happen for several reasons. Mobile device users are going to be greatly impacted by their connection speed (3G/4G/5G, Wi-Fi, etc.). Another reason can be how the page styles are loaded. Mobile devices render CSS differently compared to desktop devices, such as resizing images and considerations around viewport/screen sizes. Not to forget, mobile devices simply aren't as powerful as desktop computers.
```

Comment:
```suggestion
When running a speed test for a mobile device and a desktop device, you could notice quite the difference in performance results. This can happen for several reasons. Mobile device users are going to be greatly impacted by their connection speed (3G/4G/5G, Wi-Fi, etc.). Another reason can be how the page styles are loaded. Mobile devices render CSS differently compared to desktop devices, such as resizing images and considerations around viewport/screen sizes. Not to forget, mobile devices often are not as powerful as desktop computers.
```